embedded_fab/painters.dart

```diff
@@ -0,0 +1,250 @@
+            width: _kBumpRadius,
+          );
+
+          _silhouettePath = Path()
+            ..moveTo(0, 36)
+            ..lineTo(size.width * .5 - _kBumpRadius * .5 + 2, 24)
+            ..addArc(
+              rect,
+              degToRad(200),
+              degToRad(138),
+            )
+            ..lineTo(size.width, 36)
+            ..lineTo(size.width, size.height)
+            ..lineTo(0, size.height)
+            ..lineTo(0, 36);
+
+          _fabPath = Path()
+            ..addArc(
+              rect.shift(Offset(0, -11)),
+              degToRad(0),
+              degToRad(360),
+            );
+          break;
+        }
+      case ShapeType.Cut:
+        {
+          _silhouettePath = GoogleBabShape().getOuterPath(
+              Offset(0, 36) & size,
+              Rect.fromCenter(
+                center: Offset(size.width * .5, 24),
+                width: _kFABRadius * 2 + _kNotchPadding,
+                height: _kFABRadius * 2 + _kNotchPadding,
+              ));
+          _fabPath = Path()
+            ..addOval(Offset(size.width * .5 - _kFABRadius, 24 - _kFABRadius) &
+                Size.fromRadius(_kFABRadius));
+          break;
+        }
+      case ShapeType.Flat:
+        {
+          _silhouettePath = Path()..addRect(Offset(0, 36) & size);
+          _fabPath = Path()
+            ..addOval(Rect.fromCenter(
+                center: Offset(size.width / 2, 62),
+                width: _kMiniFABRadius * 2,
+                height: _kMiniFABRadius * 2));
+          break;
+        }
+    }
+
+    Path silhouetteShadowPath = Path.from(_silhouettePath)
+        .transform(Matrix4.translationValues(0, -5, 0).storage);
+
+    canvas.drawShadow(silhouetteShadowPath, Colors.black87, 2, false);
+
+    canvas.drawPath(_silhouettePath, Paint()..color = Colors.white);
+
+    if (hasShadow) {
+      Path shadowPath = Path.from(_fabPath)
+          .transform(Matrix4.translationValues(0, -4, 0).storage);
+      canvas.drawShadow(shadowPath, Colors.black87, 4, true);
+    }
+
+    canvas.drawPath(
+        _fabPath,
+        Paint()
+          ..color = fillColor
+          ..style = PaintingStyle.fill);
+
+    if (hasStroke) {
+      canvas.drawPath(
+          _fabPath,
+          Paint()
+            ..color = strokeColor
+            ..strokeWidth = 2
+            ..style = PaintingStyle.stroke);
+    }
+  }
+
+  @override
+  bool shouldRepaint(CustomPainter oldDelegate) {
+    return true;
+  }
+
+  @override
+  bool hitTest(Offset position) {
+    if (shape == ShapeType.Flat) return false;
+
+    return _fabPath.getBounds().contains(position);
+  }
+}
+
+class PlusPainter extends CustomPainter {
+  PlusPainter({
+    this.iconSize = 36,
+    this.yOffset = 0,
+    this.color,
+  });
+
+  final double iconSize;
+  final double yOffset;
+  final Color color;
+
+  // These are based on the original branded icon asset. The asset size is
+  // 36 x 36, the thickness of the plus is 4 and the padding is 6.
```

Comment:
Consider removing this comment, I don't think its relevant here.

Reply:
Done